core/src/commonMain/kotlin/com/powersync/attachments/sync/SyncingService.kt

```diff
@@ -0,0 +1,278 @@
+package com.powersync.attachments.sync
+
+import co.touchlab.kermit.Logger
+import com.powersync.PowerSyncException
+import com.powersync.attachments.Attachment
+import com.powersync.attachments.AttachmentContext
+import com.powersync.attachments.AttachmentService
+import com.powersync.attachments.AttachmentState
+import com.powersync.attachments.LocalStorage
+import com.powersync.attachments.RemoteStorage
+import com.powersync.attachments.SyncErrorHandler
+import com.powersync.utils.throttle
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.IO
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.buffer
+import kotlinx.coroutines.flow.merge
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+/**
+ * Service used to sync attachments between local and remote storage
+ */
+public class SyncingService(
+    private val remoteStorage: RemoteStorage,
+    private val localStorage: LocalStorage,
+    private val attachmentsService: AttachmentService,
+    private val getLocalUri: suspend (String) -> String,
+    private val errorHandler: SyncErrorHandler?,
+    private val logger: Logger,
+    private val syncThrottle: Duration = 5.seconds,
+) {
+    private val scope = CoroutineScope(Dispatchers.IO)
```

Comment:
The parent scope should also be configurable here.

Reply:
This is configurable now